client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -39,19 +39,21 @@ class ProductCreate extends React.Component {
 	componentDidMount() {
 		const { product, siteId } = this.props;
 
-		if ( ! product ) {
-			this.props.editProduct( null, {
-				type: 'simple'
-			} );
-		}
-
 		if ( siteId ) {
```

Comment:
I double checked to make sure, but `componentWillReceiveProps` handles this when the siteId comes in.